### ISSUE TITLE
Change index.php to index2.php help required

### ISSUE THREAD

**Issue by Invictaz** [albertobeta/PodcastGenerator#121]
I would like to use a different name for index.php in Podcastgenerator.
I would like to change this to index2 instead of index
How do I do this?

====

I changed some of the links in index2.php to:

`<a href="'.$url.'index2.php?p=archive&amp;cat=all"`

I seem to cannot change the link "go to podcast archive" on that one.

I editted this:

```
$PG_mainbody .= '<div style="clear:both;"><p><a href="'.$url.'index2.php?p=archive&amp;cat=all"><i class="fa fa-archive"></i> '._("Go to episodes archive").'</a></p></div>';}
```

However, index2.php?p=home still refers to index.php?p=archive&amp;cat=all instead of index**2**.php?p=archive&amp;cat=all. 

So it works on index2.php but not on the ?p=home

There is also the function of 

```
// Home page (with no ?p= in GET)
else { // if no p= specifies, e.g. just index.php with no GET

```
How do I use this feature? Do I need to change other values to have a successfull rename to index2.php?

I also changed it in core\feedgenerate.php but also core\ftpfeature needs to be changed. Maybe also some others?

**Comment by Invictaz** [comment]
This is not needed. If you just rename index to index2.php and change it in config.php the full url to 
`$url = "https://www.w00fer.nl/podcast/index2.php"; // Complete URL of the script (Trailing slash REQUIRED)`

There seems to be a trailing slash required but for me (so far) it works fine without.

**Comment by Invictaz** [comment]
Still does not fully work. Feed is messed up and feed and iTunes buttons don't work anymore

@albertobeta can you help me out?

**Comment by Invictaz** [comment]
@emilengler Why do you close this? Issue is still not solved :(

**Comment by emilengler** [comment]
@Invictaz Hello, sorry for the close.
I bulked closed all issues because 3.0 will be released tomorrow. The old issues would be unnecessary ballast.
According to your problem I don't think this can be done by ease, sorry :/
Maybe you could install PG to a different directory (subdirectory) if you need the index.php

**Comment by emilengler** [comment]
Also a modification to your webserver conf might be helpful.
Anyway, I will work on this feature in 3.1, promised :)

**Comment by Invictaz** [comment]
@emilengler Define "unneccesary ballast". Not only for convenience but also for security reasons it is quite handy to change it to a different pagename. I can come up with additional reasons. 

What do you mean with webserver config? 3.1 is still years away :(

The only solution I thought of was adding an extra button next to the RSS button. But after days fiddling with the style sheets and .php pages I cannot find where to build an extra button. 

The structure is now that I need a front door before my podcast page, with additional information and if you click on a button you end up on the podcast page. There are two solutions

-Ability to change Podcastgen to index2.php
-Add a button on the index of Podcastgen to the front door page. Not ideal but ok.

Putting it into a different directy for example url.com/podcast (front door) and /podcast2/ (actual podcast thing) might be possible. However since all my users need to update their feeds again AND static download links might not work anymore this is something I would like to avoid.

**Comment by Invictaz** [comment]
@emilengler Changing the index to index2 in various pages breaks a lot of stuff. I now have downloads working but not everything.

**Comment by emilengler** [comment]
> Define "unneccesary ballast". Not only for convenience but also for security reasons it is quite handy to change it to a different pagename. I can come up with additional reasons.

This was meant in terms of all old issues. Because 3.0 will replace 2.7 it isn't worth giving support for 2.7 anymore.

> What do you mean with webserver config? 3.1 is still years away :(

3.1 will come on the 31th of May 2020

> The only solution I thought of was adding an extra button next to the RSS button. But after days fiddling with the style sheets and .php pages I cannot find where to build an third button.

If you have some html knowledge you can add it to the theme, fortunately PG 3 has a feature for that :D. In 2.7 you edit `themes/default_modern/index.htm` line `58`. There you have the buttons. However as I just look at it it seems a bit complicated to change the feed path.

> Putting it into a different directy for example url.com/podcast (front door) and /podcast2/ (actual podcast thing) might be possible. However since all my users need to update their feeds again AND static download links might not work anymore this is something I would like to avoid.

Do you have shell access (command line access)? If so I could help you to set it up so users don't need to update their feed with linking the old location to the feed file.

Sorry for praising 3.0 so hard and abandoning 2.7. Honestly I just have a limited knowledge on 2.7 as I haven't made it.

**Comment by Invictaz** [comment]
@emilengler Unfortunately no shell access. Only FTP. 
I will upgrade to 3.0 and add the button. Can you tell me where?

The feed path needs the index.php because if you change it to index2.php then it uses 

https://www.site.com/podcast/index2.phpmedia/music.mp3

instead of /podcast/media/music.mp3

So with adding a button to the front door, the issue is only mitigated not solved.

**Comment by emilengler** [comment]
>  I will upgrade to 3.0 and add the button. Can you tell me where the feature is in the options list?

The last entry in the themes sections

**Comment by Invictaz** [comment]
Okay waiting for 3.0 :-)

**Comment by emilengler** [comment]
3.0 just came out :-)
https://github.com/albertobeta/PodcastGenerator/releases/tag/v3.0

**Comment by Invictaz** [comment]
@emilengler Congratulations. Will update after backup is done.

**Comment by emilengler** [comment]
Implemented!
Will get released with 3.1
